Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/LocationUpdateService.java

```diff
@@ -368,8 +374,142 @@ private void updateDriverStatus(Boolean status) {
         });
     }
 
+    private String getValueFromStorage(String k) {
+        SharedPreferences sharedPreff = this.getSharedPreferences(this.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+        String val = sharedPreff.getString(k, null);
+        return val;
+    }
+
+    private JSONObject getLatLng(Double lat , Double lng) throws JSONException {
+        JSONObject latLng = new JSONObject();
+        latLng.put("lat", lat);
+        latLng.put("lon", lng);
+        return latLng;
+    }
+
+    private void getRoute(Double startLat, Double startLng, Double endLat, Double endLng) {
+        StringBuilder result = new StringBuilder();
+
+        String regToken = getValueFromStorage("REGISTERATION_TOKEN");
+        String baseUrl = getValueFromStorage("BASE_URL");
+        String version = getValueFromStorage("VERSION_NAME");
+
+        if(regToken == null || baseUrl == null || version == null) {
+            return;
+        }
+        try {
+
+            String rideId = getValueFromStorage("RIDE_ID");
+            String url = baseUrl+ "/" + rideId + "/route";
+            HttpURLConnection connection = (HttpURLConnection) (new URL(url).openConnection());
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "application/json");
+            connection.setRequestProperty("token", regToken);
+            connection.setRequestProperty("x-client-version", version);
+
+            connection.connect();
+            int respCode = connection.getResponseCode();
+            InputStreamReader respReader;
+            if (respCode == 200) {
+                respReader = new InputStreamReader(connection.getInputStream());
+                BufferedReader in = new BufferedReader(respReader);
+                String inputLine;
+                while ((inputLine = in.readLine()) != null) {
+                    result.append(inputLine);
+                }
+                JSONObject res = new JSONObject(String.valueOf(result));
+                JSONArray pathLatLng = (JSONArray) res.get("points");
+                rideWaypoints = String.valueOf(pathLatLng);
+            } else {
+                Bundle params = new Bundle();
```

Comment:
Until `:rideId/route` is merged and deployed to PROD. We have to keep a fallback where the API called should be `trip/route` 

So, In else can you handle the error code? If it's 404 NOT_FOUND we can call the `trip/route`